app/src/main/java/org/fossasia/openevent/widget/BookmarkWidgetRemoteViewsService.java

```diff
@@ -0,0 +1,158 @@
+package org.fossasia.openevent.widget;
+
+import android.annotation.TargetApi;
+import android.appwidget.AppWidgetManager;
+import android.content.ComponentName;
+import android.content.Intent;
+import android.database.MatrixCursor;
+import android.os.Build;
+import android.os.Bundle;
+import android.widget.AdapterView;
+import android.widget.RemoteViews;
+import android.widget.RemoteViewsService;
+
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.dbutils.DbSingleton;
+import org.fossasia.openevent.utils.ISO8601Date;
+
+import java.util.ArrayList;
+
+/**
+ * Created by opticod(anupam) on 10/1/16.
+ */
+@TargetApi(Build.VERSION_CODES.HONEYCOMB)
+public class BookmarkWidgetRemoteViewsService extends RemoteViewsService {
+
+    static final int INDEX_BOOKMARK_ID = 0;
+    static final int INDEX_BOOKMARK_TITLE = 1;
+    static final int INDEX_BOOKMARK_START_TIME = 2;
+    static final int INDEX_BOOKMARK_END_TIME = 3;
+    ArrayList<Integer> bookmarkedIds;
+    private int LESS_DETAIL_SIZE = 300;
+
+    @Override
+    public RemoteViewsFactory onGetViewFactory(Intent intent) {
+
+        final AppWidgetManager appWidgetManager = AppWidgetManager.getInstance(this);
+        int[] appWidgetIds = appWidgetManager.getAppWidgetIds(new ComponentName(this,
+                BookmarkWidgetProvider.class));
+        final int appWidgetId = appWidgetIds[0];
+
+        return new RemoteViewsFactory() {
+            private MatrixCursor data = null;
+
+            @Override
+            public void onCreate() {
+            }
+
+            @Override
+            public void onDataSetChanged() {
+                if (data != null) {
+                    data.close();
+                }
+                DbSingleton dbSingleton = DbSingleton.getInstance();
+                try {
+                    bookmarkedIds = dbSingleton.getBookmarkIds();
+
+                    String[] columns = new String[]{"id", "title", "startTime", "endTime"};
```

Comment:
avoid hard coding, try to see if you can get as a constant or a pre-filled array.